numpy/f2py/crackfortran.py

```diff
@@ -289,22 +290,16 @@ def undo_rmbadname(names):
     return [undo_rmbadname1(_m) for _m in names]
 
 
-def getextension(name):
-    i = name.rfind('.')
-    if i == -1:
-        return ''
-    if '\\' in name[i:]:
-        return ''
-    if '/' in name[i:]:
-        return ''
-    return name[i + 1:]
-
 is_f_file = re.compile(r'.*\.(for|ftn|f77|f)\Z', re.I).match
```

Comment:
This looks fine, anyone from the "f2py team" want to look over and merge it?

Does `is_f_file` need to remain because it is effectively public API?